saleor/graphql/attribute/mutations/attribute_bulk_update.py

```diff
@@ -0,0 +1,636 @@
+    get_duplicated_values,
+)
+from ...core.validators import validate_one_of_args_is_in_mutation
+from ...plugins.dataloaders import get_plugin_manager_promise
+from ..enums import AttributeTypeEnum
+from ..types import Attribute
+from .attribute_bulk_create import DEPRECATED_ATTR_FIELDS, clean_values
+from .attribute_update import AttributeUpdateInput
+
+
+class AttributeBulkUpdateResult(BaseObjectType):
+    attribute = graphene.Field(Attribute, description="Attribute data.")
+    errors = NonNullList(
+        AttributeBulkUpdateError,
+        required=False,
+        description="List of errors occurred on update attempt.",
+    )
+
+    class Meta:
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+
+def get_results(
+    instances_data_with_errors_list: list[dict], reject_everything: bool = False
+) -> list[AttributeBulkUpdateResult]:
+    return [
+        AttributeBulkUpdateResult(
+            attribute=None if reject_everything else data.get("instance"),
+            errors=data.get("errors"),
+        )
+        for data in instances_data_with_errors_list
+    ]
+
+
+class AttributeBulkUpdateInput(BaseInputObjectType):
+    id = graphene.ID(description="ID of an attribute to update.", required=False)
+    external_reference = graphene.String(
+        description="External ID of this attribute.", required=False
+    )
+    fields = AttributeUpdateInput(description="Fields to update.", required=True)
+
+    class Meta:
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+
+class AttributeBulkUpdate(BaseMutation):
+    count = graphene.Int(
+        required=True,
+        description="Returns how many objects were updated.",
+    )
+    results = NonNullList(
+        AttributeBulkUpdateResult,
+        required=True,
+        default_value=[],
+        description="List of the updated attributes.",
+    )
+
+    class Arguments:
+        attributes = NonNullList(
+            AttributeBulkUpdateInput,
+            required=True,
+            description="Input list of attributes to update.",
+        )
+        error_policy = ErrorPolicyEnum(
+            required=False,
+            description="Policies of error handling. DEFAULT: "
+            + ErrorPolicyEnum.REJECT_EVERYTHING.name,
+        )
+
+    class Meta:
+        description = "Updates attributes." + ADDED_IN_315 + PREVIEW_FEATURE
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+        error_type_class = AttributeBulkUpdateError
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.ATTRIBUTE_UPDATED,
+                description=(
+                    "An attribute was updated. "
+                    "Optionally called when  new attribute value was created or "
+                    "deleted."
+                ),
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.ATTRIBUTE_VALUE_CREATED,
+                description="An attribute value was created.",
```

Comment:
You could also place here a info that the events are optionally called. Right? VALUE_CREATED will be called when you add new values, VALUE_DELETED will be called when you delete values. Both events are optionally, I think it is worth to mention.

Reply:
Added.